### PR TITLE
Support custom instance_types

### PR DESCRIPTION
**Pull request by jameswnl**
Allow user to add instance_types that MIQ doesn't include. These are types that added by AWS after MIQ's code freeze for release.

### CODE REVIEW

**Comment by jameswnl** [comment]
@miq-bot add_labels enhancement

File: lib/tasks_private/instance_types.rake [hunk 1]
```diff
@@ -1,12 +1,12 @@
-desc "Create new instance_types.rb"
-task :instance_types => [:environment] do
+desc("Create new instance_types.rb")
+task(:instance_types => [:environment]) do
```
Comment (Fryguy) [inline lib/tasks_private/instance_types.rake]: Don't add the parens in...these are DSL methods.  If anything we should change the codeclimate.yml if that's the reason for this.  cc @bdunne

Comment (bdunne) [reply lib/tasks_private/instance_types.rake]: A section for the rake gem under https://github.com/ManageIQ/guides/blob/master/.rubocop_base.yml#L75 would be great.  @jameswnl do you want to add that or should I?

Comment (jameswnl) [reply lib/tasks_private/instance_types.rake]: @Fryguy updated!
@bdunne yes, in this [PR](https://github.com/ManageIQ/guides/pull/249)

File: lib/tasks_private/instance_types.rake [hunk 2]
```diff
@@ -190,14 +190,20 @@ EOT
 
   new_instance_types_file << output_instances.map { |i| format_instance(i) }.join("\n")
   new_instance_types_file << <<EOT
-  }
+  }.freeze
+
+  def self.instance_types
```
Comment (bdunne) [inline lib/tasks_private/instance_types.rake]: Is this copied from above?  Can we use that method instead of duplicating it?

Comment (jameswnl) [reply lib/tasks_private/instance_types.rake]: You mean the `def self.instance_types` method in `instance_types.rb`?  
The `instance_types.rb` is the output of this rake task

Comment (Fryguy) [reply lib/tasks_private/instance_types.rake]: I find it weird in general that this generates source code.  I feel this task should just generate a YAML file, which is processed on first access by the code.  Then this "duplication" wouldn't exist either.

Comment (jameswnl) [reply lib/tasks_private/instance_types.rake]: I feel strongly the same as well.

My plan is to finish this task first for 4.6 and then revisit this.

**Comment by bronaghs** [comment]
Going to merge this for now, but we will improve this in the near future to generate a JSON or YAML output file instead of a .rb file.

**Comment by bronaghs** [comment]
@jameswnl - Can this be backported to FINE?

**Comment by jameswnl** [comment]
@bronaghs yes, I think so.

**Comment by jameswnl** [comment]
@miq-bot remove_labels fine/no
@miq-bot add_labels fine/yes

@simaishi this should be backportable. let me know if you need help. thanks

**Comment by simaishi** [comment]
@bronaghs please add BZ link.

**Comment by bronaghs** [comment]
@simaishi - https://bugzilla.redhat.com/show_bug.cgi?id=1557025

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit 27eb32c874b4c877f200b81b0e141b73a2703287
Author: Bronagh Sorota <bsorota@redhat.com>
Date:   Mon Sep 18 14:35:06 2017 -0400

    Merge pull request #289 from jameswnl/instancetypes
    
    Support custom instance_types
    (cherry picked from commit 5d3a5f8950878e75d9cbef6255213b9c04e0aaa9)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1557025
```

**Pull request merged**